src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -19,48 +24,70 @@ export type UseSpaceBySlugResult = {
  * @param spaceSlug
  * @returns
  */
-export const useSpaceBySlug = (spaceSlug?: string): UseSpaceBySlugResult => {
+export const useSpaceBySlug = (
+  worldSlug?: WorldSlug,
+  spaceSlug?: SpaceSlug
+): UseSpaceBySlugResult => {
   const firestore = useFirestore();
 
   const spacesRef = firestore
     .collection("venues")
-    .where("slug", "==", spaceSlug ?? "")
+    .where("slug", "==", spaceSlug ?? EMPTY_SPACE_SLUG)
     .withConverter(withIdConverter<AnyVenue>());
 
   // Note: Avoid using the option 'initialData' because it will make status always return 'success'
-  const { data: spaces, status } = useFirestoreCollectionData<WithId<AnyVenue>>(
-    spacesRef
-  );
+  const { data: spaces, status: spaceStatus } = useFirestoreCollectionData<
+    WithId<AnyVenue>
+  >(spacesRef);
 
-  const isSpaceLoaded = status !== "loading";
+  const worldsRef = firestore
+    .collection("worlds")
+    .where("slug", "==", worldSlug ?? EMPTY_WORLD_SLUG)
+    .withConverter(withIdConverter<World>());
+  const { data: worlds, status: worldStatus } = useFirestoreCollectionData<
+    WithId<World>
+  >(worldsRef);
 
-  const space = spaces?.[0];
+  const isSpaceLoaded = spaceStatus !== "loading";
+  const isWorldLoaded = worldStatus !== "loading";
+
+  const world = worlds?.[0];
+  if (!isWorldLoaded || !isSpaceLoaded) {
+    return {
+      world: undefined,
+      space: undefined,
+      spaceId: undefined,
+      isLoaded: false,
+    };
+  }
 
-  if (!space && isSpaceLoaded) {
+  if (!world) {
     return {
+      world: undefined,
       space: undefined,
       spaceId: undefined,
       isLoaded: true,
-      error: `Space with the following slug: ${spaceSlug} does not exist.`,
+      error: `World with the following slug: ${worldSlug} does not exist.`,
     };
   }
 
-  if (spaces?.length > 1) {
-    Bugsnag.notify(
-      `Multiple spaces have been found with the following slug: ${spaceSlug}.`,
-      (event) => {
-        event.severity = "warning";
-        event.addMetadata("hooks::useSpaceBySlug", {
-          spaceSlug,
-          spaces,
-        });
-      }
-    );
+  const space = spaces.find(
+    (candidateSpace) => candidateSpace.worldId === world.id
+  );
```

Comment:
This still removes the ability to be notified if for some reason there is a double booking in the DB.

Reply:
Ahhh. Yes. Good spot.
Resolved in d980d71a